app/models/button_maker.rb

```diff
@@ -51,242 +44,17 @@ def set_item_properties
-      call_number: @call_number,
-      collection: @collection,
-      library: @library,
-      title: @title,
-      year: @year,
-      volume: @volume
-    ).url_for_scan
-  end
-
-  def url_for_ill
-    return unless @oclc_number.present?
-    "https://mit.worldcat.org/search?q=no%3A#{@oclc_number}"
-  end
-
-  # ~~~~~~~~ Utility functions needed to determine PDF scan eligibility ~~~~~~~~
-  def call_number_valid_for_scan?
-    !@call_number.start_with?('ARCH DRAW', 'ATLAS', 'AUDIO', 'AUDTAPE', 'CD',
-                              'CDROM', 'DSKETTE', 'DVD', 'DVDROM', 'FICHE',
-                              'FILM', 'FOLIO', 'INDEX', 'MAP', 'MFILM',
-                              'OVRSIZE', 'RECORD', 'REGULAR', 'SCORE', 'SMALL',
-                              'THESIS', 'USB DRIVE', 'VDISC', 'VIDEO')
-  end
-
-  def z30status_valid_for_scan?
-    [
-      '60 Day Loan',
-      'One Week Loan',
-      '1 Week No Renew',
-      'OCC 60',
-      'LSA 7',
-      'LSA 60',
-      'LSA Use Only'
-    ].include? @z30status
-  end
-
-  def collection_valid_for_scan?
-    [
-      'Stacks',
-      'Journal Collection',
-      'Off Campus Collection',
-      'Science Journals',
-      'Humanities Journals',
-      'Impulse Borrowing Display',
-      'Browsery',
-      'Graphic Novel Collection',
-      'Travel Collection'
-    ].include? @collection
-  end
-
-  def status_valid_for_scan?
-    ['In Library', 'New Books Displ'].include?(@status)
-  end
-
-  def library_valid_for_scan?
-    ['Physics Dept. Reading Room', 'Rotch Visual Collections',
-     'Institute Archives'].exclude? @library
-  end
-
-  # We don't scan certain engineering standards. See
-  # https://wikis.mit.edu/confluence/x/5igEBw for a link to the relevant
-  # spreadsheet.
-  def unscannable_standard?
-    @scan == 'false'
-  end
-
   # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Misc utilities ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-  # Items with these process & item statuses may be put on hold/recalled.
-  def hold_recallable?
-    return false if @on_reserve || @library == 'Physics Dept. Reading Room'
-    [
-      # Items with these status codes may be requested from any library,
-      # except the Annex.
-      %w(01 03 05 12 19 21).include?(@z30status_code) &&
-        @library != 'Library Storage Annex',
-
-      # Items with status code 23 may be requested from only some libraries.
-      @z30status_code == '23' && ['Barker Library',
-                                  'Dewey Library',
-                                  'Hayden Library',
-                                  'Rotch Library',
-                                  'Library Storage Annex'].include?(@library),
-
-      # The Annex is special.
-      %w(13 14 15 56 57).include?(@z30status_code) &&
-        @library == 'Library Storage Annex'
-    ].any?
-  end
-
   # We won't generally ILL items that fit these criteria - we want them to be
   # placed on hold if possible.
-  def should_you_get_it_here?
+  def available_locally?
```

Comment:
I do think that a somewhat different name is better, per discussion on the other PR - the key thing is that it's available both here _and now_ (as availability is very time-dependent). `available_here_now?` ?

Reply:
👍